### ISSUE TITLE
Format can causes compile error if function has multi line comments

### ISSUE THREAD

**Issue by RyosukeFukatani** [pinterest/ktlint#2355]
## Expected Behavior

Input
```kotlin
fun zero():
        Float
// comment1
// comment2
{
    return 0.0f
}
```

Expected result
```kotlin
fun zero(): Float {
// comment1
// comment2
    return 0.0f
}
```
or other compilable code.

## Observed Behavior
Unfortunately we got non-compilable code.

```kotlin
fun zero(): Float
// comment1 { // comment2
        return 0.0f
    }

```

## Your Environment

* Version of ktlint used: 1.0.1
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):Gradle
* Version of Gradle used (if applicable): 7.4.2
* Operating System and version: mac OS 14.0

**Comment by paul-dingemans** [comment]
Tnx fore the report. This is clearly a bug. I am not sure whether I can solve it within reasonable bounds. But least that I can do, is to disallow this particular location for a comment.

I don't know your usage of this way of commenting, but I would most likely have put the comment somewhere else.

**Comment by RyosukeFukatani** [comment]
Thanks!
Actually, we consider this commenting style to be an unusual one and do not intend to continue using it.
In general, the following styles are considered preferable.

```kotlin
// comment1
// comment2
fun zero(): Float {
    return 0.0f
}
```
